src/server/utils/timeout.js

```diff
@@ -22,3 +22,6 @@ export default async function requestTimeout(millis, timeoutReason = null) {
 
   await timeout(millis, errorMessage)
 }
+
+Promise.timeout = (promises, timeout, timeoutReason) =>
```

Comment:
1. do not need to make it 'universal', let it just receives some promise. If someone would need pass array of promises, he would wrap it to Promise.all by himself
2. we probably may want to use it with retry(). So let pass not a promise itself but the async function returning promise
3. Modifying existing JS objects is a kind of the bad design. it should be used only if we want to polyfill some spec which doesn't exists on the JS engine we're using.

Following the pt 1-3 i suggest to usethe implementation below:

```javascript
export const runWithTimeout(asyncFn, millis, timeoutReason = null) =>
  Promise.race([asyncFn(), requestTimeout(millis, timeoutReason)])
```